staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package validation validates EncryptionConfiguration.
+package validation
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/config"
+)
+
+const (
+	moreThanOneElementErr = "more than one provider specified in a single element, should split into different list elements"
+)
+
+var (
+	aesKeySizes              = []int{16, 24, 32}
+	secretBoxKeySizes        = []int{32}
+	root                     = field.NewPath("EncryptionConfiguration")
+	resource                 = root.Child("Resource")
+	keyLenErrFmt             = "secret is not of the expected length, got %d, expected one of %v"
+	unsupportedSchemeErrFmt  = "unsupported scheme %q for KMS provider, only unix is supported"
+	atLeastOneRequiredErrFmt = "at least one %s is required"
+	mandatoryFieldErrFmt     = "%s is a mandatory field for a %s"
+	base64EncodingErr        = "Secrets must be base64 encoded"
+	zeroOrNegativeErrFmt     = "%s should be a positive value"
+)
+
+// ValidateEncryptionConfiguration validates a v1.EncryptionConfiguration.
+func ValidateEncryptionConfiguration(c *config.EncryptionConfiguration) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if c == nil {
+		allErrs = append(allErrs, field.Required(root, "EncryptionConfiguration can't be nil"))
+		return allErrs
+	}
+
+	if len(c.Resources) == 0 {
+		allErrs = append(allErrs, field.Required(root, fmt.Sprintf(atLeastOneRequiredErrFmt, resource)))
+		return allErrs
+	}
+
+	for i, conf := range c.Resources {
+		r := resource.Index(i).Child("Resources")
+		p := resource.Index(i).Child("Providers")
+
+		if len(conf.Resources) == 0 {
+			allErrs = append(allErrs, field.Required(r, fmt.Sprintf(atLeastOneRequiredErrFmt, r)))
+		}
+
+		if len(conf.Providers) == 0 {
+			allErrs = append(allErrs, field.Required(p, fmt.Sprintf(atLeastOneRequiredErrFmt, p)))
+		}
+
+		for j, provider := range conf.Providers {
+			path := p.Index(j)
+			found := false
+
+			if provider.KMS != nil {
+				allErrs = append(allErrs, validateKMSConfiguration(provider.KMS, path)...)
+				found = true
+			}
+
+			if provider.AESGCM != nil {
+				if found {
+					allErrs = append(allErrs, field.Invalid(path, provider, moreThanOneElementErr))
+				} else {
+					allErrs = append(allErrs, validateKeys(provider.AESGCM.Keys, path, aesKeySizes)...)
```

Comment:
pass `path.Child("AESGCM").Child("Keys")` instead of `path`

Reply:
Done.